ui/src/containers/Search/index.tsx

```diff
@@ -1,28 +1,48 @@
 import React from 'react';
 import { useHistory } from 'react-router-dom';
-import { useObserver } from 'mobx-react';
 import { useHotkeys } from 'react-hotkeys-hook';
-import { TextInput } from '@patternfly/react-core';
 import { useMst } from '../../store/root';
-import { useDebounce } from '../../utils/useDebounce';
+import { observer } from 'mobx-react';
+import { TextInput } from '@patternfly/react-core';
+import { useDebouncedCallback } from 'use-debounce';
 import { assert } from '../../store/utils';
 import './Search.css';
 
-const Search: React.FC = () => {
+const Search: React.FC = observer(() => {
+  const { resources } = useMst();
+  const [value, setValue] = React.useState(resources.search);
+
+  React.useEffect(() => {
+    setValue(resources.search);
+  }, [resources.search]);
+
+  const inputRef = React.useRef<HTMLInputElement>(null);
   useHotkeys('/', () => {
     // Without timeout the hotkey will be sent to select input
     setTimeout(() => {
-      const currentInput = inputRef.current;
-      assert(currentInput);
-      currentInput.focus();
+      assert(inputRef.current);
+      inputRef.current.focus();
     }, 20);
   });
 
-  const { resources } = useMst();
+  /* Debounce callback
+     This function delay the invoking func for 500ms after
+     when users stop searching
+  */
+  const debounced = useDebouncedCallback(
+    (value: string) => {
+      resources.setSearch(value);
+    },
+    // Delay in ms
+    500
```

Comment:
Why the delay in search is 500 ??
Where the user data which he has typed for searching will be stored till 500ms are done? Should that be stored in the localstorage ??

Reply:
it's a custom delay , I kept it 500 ms (I thought 500 ms is fine ,if you think any other time would be fine we can change it).
search data is fetching from the `search box` itself

Reply:
Does it depend on the number of resources in the store ??

Reply:
nope , it's a delay to used for debounce (any function  will be execute only after that delay), In our case store get update (filter function to be called) only just after 500 ms when user stop searching